File: LayoutTests/http/tests/performance/paint-timing/resources/slowscript.py
#!/usr/bin/env python3

import os
import sys
import time
from urllib.parse import unquote

query_string = os.environ.get('QUERY_STRING', '')
query_arr = query_string.split('&')
query = {}
for part in query_arr:
    part = part.split('=')
    if len(part) > 1:
        query[part[0]] = []
        for sub_part in part[1:]:
            query[part[0]].append(unquote(sub_part))

delay = query.get('delay', [None])[0]
script = query.get('script', [''])[0]

if delay is not None:
    time.sleep(float(delay))
sys.stdout.write(
    'Cache-Control: no-cache, must-revalidate\r\n'
    'Content-Type: text/javascript\r\n\r\n{}'.format(script)
)

File: LayoutTests/http/tests/resources/slow-script.pl
#!/usr/bin/perl -w

use strict;
use CGI;
use Time::HiRes qw(sleep);

my $cgi = new CGI;
my $delay = $cgi->param('delay');
$delay = 5000 unless $delay;

# flush the buffers after each print
select (STDOUT);
$| = 1;

print "Content-Type: application/javascript\n";
print "Expires: Thu, 01 Dec 2003 16:00:00 GMT\n";
print "Cache-Control: no-store, no-cache, must-revalidate\n";
print "Pragma: no-cache\n";
print "\n";

sleep $delay / 1000;


